app/code/Magento/Swatches/Controller/Ajax/Media.php

```diff
@@ -56,21 +60,33 @@ public function __construct(
     public function execute()
     {
         $productMedia = [];
+        $searchCriteria = $this->searchCriteria->addFilter('additional_data', '%update_product_preview_image%', 'like')->create();
```

Comment:
I cannot see the reference to specific attribute. Is this solution applicable in case with two swatch attributes, when the first swatch attribute should update the image and the second shouldn't?

Reply:
If there are two swatch attributes when the first swatch attribute should update the image and second shouldn't then swatch image not change because one swatch attribute has update_product_preview_image = No

Reply:
I think that clicking the swatch of the first attribute, in this case, should still update the images because it has `update_product_preview_image` set to Yes.

Reply:
so as per your suggestion if one attribute set update_product_preview_image to Yes then image change.
like size and color both have the attribute value update_product_preview_image now if any one set to yes then product image change.
if I am wrong then correct me.

Reply:
My understanding is the following:
* We have two swatch attributes (i.e. colour and size as in sample data) - there are two rows of swatches for a product on frontend.
* Colour attribute is configured `update_product_preview_image=yes`
* Size is configured `update_product_preview_image=no`

Then clicking any swatch in the first row (selecting colour) should change the product image, while clicking any swatch in the second row (selecting size) should not change the product image.

Reply:
Now if we have color set update_product_preview_image=yes And Size set update_product_preview_image=No 
Now if click on color swatch then images change and if we click on size swatch then image not change.

Reply:
Do you mean it works that way on the current 2.3-develop ?

Reply:
ok, I  have change my code as per your suggestion.

Reply:
Hello @sivaschenko 
I have debugged code and I had found in ajax we do not pass any additional parameter for so any backend we can find ajax call comes from either size swatch or color swatch so it's complected to decide to update image preview or not
the solution has two possibility 
1. like And condition if   size swatch update_product_preview_image=yes and color swatch update_product_preview_image=yes,  all swatch attribute yes then update image
2. like Or condition if size swatch update_product_preview_image=Yes and color swatch update_product_preview_image=No,  either any swatch attribute yes then  update image 
so which solution is most effective?
or any additional suggestions?

Reply:
@sivaschenko thanks for replay 
Front side, we have not additional information regarding swatch attribute additional data which is stored in searialized data so it's complected to get additional data of attribute in the front end.

Reply:
@ravi-chandra3197 is there a way to pass the required data to frontend side while initial rendering?

Reply:
I have Try another way to achieve the same functionality

Reply:
Hello @sivaschenko 
I have updated the code as per your suggestion and implement solution using swatch-renderer.js

https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js